lib/nerves_hub_link/socket.ex

```diff
@@ -133,11 +148,20 @@ defmodule NervesHubLink.Socket do
       |> assign(joined_at: nil)
       |> assign(firmware_validation_timer_pid: nil)
 
-    if config.connect_wait_for_network do
-      schedule_network_availability_check()
-      {:ok, socket}
-    else
-      {:ok, socket, {:continue, :connect}}
+    cond do
+      not config.connect ->
+        {:ok, socket}
+
+      not Client.ready_to_connect?() ->
+        schedule_establish_connection_check()
+        {:ok, socket}
+
+      config.connect_wait_for_network ->
```

Comment:
Is the assumption here that if you ever implement checks to return `false` on `ready_to_connect?`, you will skip any `wait_for_network` check?  If I've traced things right, that seems to be the behavior

Reply:
That was the original intent, but I can change this, thoughts?